pennylane/vqe/vqe.py

```diff
@@ -55,6 +55,18 @@ class Hamiltonian:
     >>> print(H)
     (0.2) [X0 Z1] + (-0.543) [Z0 H2]
 
+    The user can also provide custom observables:
+
+    >>> obs_matrix = np.array([[0.5, 1.0j, 0.0, -3j],
+                               [-1.0j, -1.1, 0.0, -0.1],
+                               [0.0, 0.0, -0.9, 12.0],
+                               [3j, -0.1, 12.0, 0.0]])
+    >>> obs = qml.Hermitian(obs_matrix, wires=[0, 1])
+    >>> obs.name = 'a'
```

Comment:
@josh146 raised the valid point that having a custom observable name might break plugin observable mappings, so perhaps it's good to keep to the default observable names